lib/components/avatar/avatar.dart

```diff
@@ -23,34 +22,35 @@ class TUIAvatar extends StatelessWidget {
   final TUIAvatarSize avatarSize;
   final TUIAvatarContent avatarContent;
   final Color? textColor;
-  final Color? foregroundColor;
   final Color? backgroundColor;
   final bool isBadged;
 
   const TUIAvatar({
     super.key,
     required this.avatarSize,
     required this.avatarContent,
-    this.textColor = TUIDefaultColors.onTertiary,
-    this.foregroundColor = TUIDefaultColors.constantLight,
-    this.backgroundColor = TUIDefaultColors.tertiary,
+    this.textColor,
+    this.backgroundColor,
     this.isBadged = false,
   });
 
   @override
   Widget build(BuildContext context) {
     Widget circleChild;
+    TUIThemeData theme = TUITheme.of(context);
+    final textColor = this.textColor ?? theme.colors.onTertiary;
+    final backgroundColor = this.backgroundColor ?? theme.colors.tertiary;
 
```

Comment:
Similar to my other comment why don't we extract colors to constant and use that instead of theme.colors.